### PR TITLE
Upgrade ArcK8sDiagnoser and fix Style, Linter issues

### PR DESCRIPTION
**Pull request by NarayanThiru**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az connectedk8s

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/connectedk8s/azext_connectedk8s/_clientproxyutils.py [hunk 1]
```diff
@@ -43,10 +43,7 @@ def close_subprocess_and_raise_cli_error(proc_subprocess, msg):
 
 
 def check_if_csp_is_running(clientproxy_process):
-    if clientproxy_process.poll() is None:
-        return True
-    else:
-        return False
+    return (clientproxy_process.poll() is None)
```
Comment (9lash) [inline src/connectedk8s/azext_connectedk8s/_clientproxyutils.py:46]: What is the expected return object of clientproxy_process.poll()?

**Review by 9lash — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/connectedk8s/azext_connectedk8s/_clientproxyutils.py (hunk 1 above)
Comment (NarayanThiru) [reply src/connectedk8s/azext_connectedk8s/_clientproxyutils.py:46]: Discussed offline

**Comment by NarayanThiru** [comment]
> Please fix CI issues

The azdev setup task is failing in all these jobs - it is happening across PRs.  Can you pl. investigate and fix this?

**Comment by NarayanThiru** [comment]
@bebound , can you please review the PR, and help merge the changes?

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 2]
```diff
@@ -62,12 +62,14 @@
 # pylint: disable=line-too-long
 
 
-def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlation_id=None, https_proxy="", http_proxy="", no_proxy="", proxy_cert="", location=None,
-                        kube_config=None, kube_context=None, no_wait=False, tags=None, distribution='generic', infrastructure='generic',
-                        disable_auto_upgrade=False, cl_oid=None, onboarding_timeout="600", enable_private_link=None, private_link_scope_resource_id=None,
-                        distribution_version=None, azure_hybrid_benefit=None, skip_ssl_verification=False, yes=False, container_log_path=None):
+def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlation_id=None, https_proxy="",
```
Comment (AllyW) [inline src/connectedk8s/azext_connectedk8s/custom.py:65]: default value change is breaking change and version should increase major number.

File: src/connectedk8s/setup.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
 
-VERSION = '1.7.3'
+VERSION = '1.7.4'
```
Comment (AllyW) [inline src/connectedk8s/setup.py:20]: 2.0.0. Please refer to extension schema: https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md

**Comment by yanzhudd** [comment]
please resolve code conflicts

File: src/connectedk8s/azext_connectedk8s/_precheckutils.py [hunk 4]
```diff
@@ -154,19 +150,22 @@ def executing_cluster_diagnostic_checks_job(corev1_api_instance, batchv1_api_ins
...
                     return
 
         chart_path = azext_utils.get_chart_path(consts.Cluster_Diagnostic_Checks_Job_Registry_Path, kube_config,
                                                 kube_context, helm_client_location,
                                                 consts.Pre_Onboarding_Helm_Charts_Folder_Name,
                                                 consts.Pre_Onboarding_Helm_Charts_Release_Name, False)
 
+        print("Step: {}: Chart path for Cluster Diagnostic Checks Job: {}".format(azext_utils.get_utctimestring(),
+                                                                                  chart_path))
+        print("Step: {}: Creating Cluster Diagnostic Checks job".format(azext_utils.get_utctimestring()))
```
Comment (yanzhudd) [inline src/connectedk8s/azext_connectedk8s/_precheckutils.py:168]: It is better to use logger instead of print function.

Comment (NarayanThiru) [reply src/connectedk8s/azext_connectedk8s/_precheckutils.py:168]: These messages are not Warnings, so didn't want to use logger.Warning.  I want these to show up even if the user runs it without --debug.  For some reason logger.Info isn't working - it doesn't output to stdout, which is why I am using print.

**Comment by NarayanThiru** [comment]
Hi folks, I have addressed the comments.  Can you please take another look?

**Comment by NarayanThiru** [comment]
@yanzhudd , @AllyW , @bebound , can you pl. help complete the review?  We have customers waiting on this change, and would greatly appreciate your help in merging the PR and releasing it.

**Review by yanzhudd — APPROVED** [review]

**Comment by NarayanThiru** [comment]
Thanks a lot for your help with merging this, @yanzhudd - really appreciate it!

**Pull request merged**